app/scripts/proactive/view/dom.js

```diff
@@ -428,6 +428,11 @@ define(
                 return;
             }
 
+            for( var key in jobVariables){
+                if(jobVariables[key].Model.indexOf('$') !== -1){
+                    jobVariables[key].referenceType = jobVariables[jobVariables[key].Model.split('$')[1]].Value;
```

Comment:
I would name this variable jobVariables[key].resolvedModel.
And I would do:
- jobVariables[key].resolvedModel=jobVariables[key].Model
- for each element pattern of the map built above:
   jobVariables[key].resolvedModel = jobVariables[key].resolvedModel.replace(pattern, replacement)